src/pages/PlanPage/subRoutes/CurrentOrgPlan/BillingDetails/EmailAddress/EmailAddress.jsx

```diff
@@ -0,0 +1,113 @@
+import { zodResolver } from '@hookform/resolvers/zod'
+import { useState } from 'react'
+import { useForm } from 'react-hook-form'
+import { useParams } from 'react-router-dom'
+import { z } from 'zod'
+
+import { useAccountDetails, useUpdateBillingEmail } from 'services/account'
+import A from 'ui/A'
+import Button from 'ui/Button'
+import Icon from 'ui/Icon'
+import TextInput from 'ui/TextInput'
+
+const emailSchema = z.object({
+  newCustomerEmail: z.coerce
+    .string()
+    .min(1, { message: 'This field has to be filled' })
+    .email({ message: 'Invalid email address' }),
+})
+
+function EmailAddress() {
+  const { provider, owner } = useParams()
+  const { data: accountDetails } = useAccountDetails({ provider, owner })
+  const [isFormOpen, setIsFormOpen] = useState(false)
+  const currentCustomerEmail =
+    accountDetails?.subscriptionDetail?.customer?.email || null
+
+  const {
+    register,
+    handleSubmit,
+    formState: { isValid, errors },
+  } = useForm({
+    defaultValues: {
+      newCustomerEmail: currentCustomerEmail,
+    },
+    resolver: zodResolver(emailSchema),
+    mode: 'onChange',
+  })
+
+  const { mutate, isLoading } = useUpdateBillingEmail({ provider, owner })
+
+  const submit = (data) => {
+    return mutate(
+      { newEmail: data?.newCustomerEmail },
+      {
+        onSuccess: () => {
+          setIsFormOpen(false)
+        },
+      }
+    )
+  }
+
+  return (
+    <div className="flex flex-col gap-2 border-t p-4">
+      <div className="flex justify-between">
+        <h4 className="font-semibold">Email address</h4>{' '}
+        {!isFormOpen && (
+          /* @ts-expect-error */
```

Comment:
Any reason this file is `jsx` instead of `tsx`?

Reply:
I changed it 👍 some ts-expect-errors thrown though